packages/core/src/llm/model/model.ts

```diff
@@ -295,10 +318,31 @@ export class MastraLLM extends MastraLLMBase {
       },
     };
 
+    let schema: z.ZodType<Z> | Schema<Z> | undefined;
+
+    if (experimental_output) {
+      this.logger.debug('[LLM] - Using experimental output', {
+        runId,
+      });
+      if (typeof (experimental_output as any).parse === 'function') {
+        schema = experimental_output as z.ZodType<Z>;
+        if (schema instanceof z.ZodArray) {
+          schema = schema._def.type as z.ZodType<Z>;
+        }
+      } else {
+        schema = jsonSchema(experimental_output as JSONSchema7) as Schema<Z>;
+      }
+    }
+
     return await streamText({
       messages,
       ...argsForExecute,
       experimental_telemetry: this.experimental_telemetry,
+      experimental_output: schema
+        ? Output.object({
+            schema,
+          })
+        : undefined,
```

Comment:
What happens if `output` and `experimental_output` are both added? Not sure if it matters

Reply:
🤷 hahaha

I'll update the docs

Reply:
Should we fix types so it cannot be done?